### PR TITLE
blob/all: add ReaderOptions to NewReader funcs

### LINKED ISSUE #765: blob: should NewReader take Options and support chunking?
Currently `blob` exposes:

-- `NewRangeReader`: takes offset, len and calls driver.
-- `NewReader`: calls `NewRangeReader` with offset=0, len=-1 (all).

To avoid reading the entire blob into memory, the caller would have to implement their own chunking.

Perhaps we should just have `NewReader`, taking `ReaderOptions`, where `ReaderOptions` has:
-- offset
-- length
-- chunk size (defaulting to something large but reasonable)

Then the concrete type would do chunking for files larger than the chunk size.

### PR DESCRIPTION
**Pull request by vangent**
Fixes #765.

### CODE REVIEW

File: blob/blob.go [hunk 1]
```diff
@@ -308,9 +308,10 @@ func (b *Bucket) As(i interface{}) bool {
 	return b.b.As(i)
 }
 
-// ReadAll is a shortcut for creating a Reader via NewReader and reading the entire blob.
+// ReadAll is a shortcut for creating a Reader via NewReader with default
+// ReaderOptions, and reading the entire blob.
```
Comment (vangent) [inline blob/blob.go]: Could also add `ReaderOptions` arg here and pass it through, WDYT?

File: README.md [hunk 2]
```diff
@@ -17,7 +17,7 @@ Imagine writing this to read from blob storage (like Google Cloud Storage or
 S3):
 
 ```go
-blobReader, err := bucket.NewReader(context.Background(), "my-blob")
+blobReader, err := bucket.NewReader(context.Background(), "my-blob", &blob.ReaderOptions{})
```
Comment (zombiezen) [inline README.md]: `nil`?
```suggestion
blobReader, err := bucket.NewReader(context.Background(), "my-blob", nil)
```

File: blob/blob.go (hunk 1 above)
Comment (zombiezen) [reply blob/blob.go]: I don't feel like we should until we know what `ReaderOptions` will contain.

File: blob/blob.go [hunk 3]
```diff
@@ -366,8 +367,8 @@ func (b *Bucket) Attributes(ctx context.Context, key string) (Attributes, error)
 // is not found by the given key, which can be checked by calling IsNotExist.
```
Comment (zombiezen) [inline blob/blob.go]: Docs (here and below) should explicitly state that `nil` options are treated the same as passing the zero value.

File: blob/blob.go [hunk 4]
```diff
@@ -505,6 +510,9 @@ type SignedURLOptions struct {
 	Expiry time.Duration
 }
 
+// ReaderOptions controls Reader behaviors.
```
Comment (zombiezen) [inline blob/blob.go]: Consider leaving a comment that it is empty for future compatibility.

File: blob/driver/driver.go [hunk 5]
```diff
@@ -22,6 +22,9 @@ import (
 	"time"
 )
 
+// ReaderOptions controls Reader behaviors.
```
Comment (zombiezen) [inline blob/driver/driver.go]: Same advice here: consider leaving a comment that it is empty for future compatibility.

**Pull request merged**